menpo/model/gmrf.py

```diff
@@ -0,0 +1,1290 @@
+        using the specified number of compnents.
+    single_precision : `bool`, optional
+        When ``True``, the GMRF's precision matrix will have `np.float32`
+        precision, else it will be `np.float64`.
+    sparse : `bool`, optional
+        When ``True``, the GMRF's precision matrix has type
+        `scipy.sparse.bsr_matrix`, otherwise it is a `numpy.array`.
+    bias : `int`, optional
+        Default normalization is by ``(N - 1)``, where ``N`` is the number of
+        observations given (unbiased estimate). If `bias` is 1, then
+        normalization is by ``N``. These values can be overridden by using
+        the keyword ``ddof`` in numpy versions >= 1.5.
+    incremental : `bool`, optional
+        This argument must be set to ``True`` in case the user wants to
+        incrementally update the GMRF. Note that if ``True``, the model
+        occupies 2x memory.
+    verbose : `bool`, optional
+        If ``True``, the progress of the model's training is printed.
+
+    Notes
+    -----
+    Let us denote a graph as :math:`G=(V,E)`, where
+    :math:`V=\{v_i,v_2,\ldots, v_{|V|}\}` is the set of :math:`|V|` vertices and
+    there is an edge :math:`(v_i,v_j)\in E` for each pair of connected vertices.
+    Let us also assume that we have a set of random variables
+    :math:`X=\{X_i\}, \forall i:v_i\in V`, which represent an abstract feature
+    vector of length :math:`k` extracted from each vertex :math:`v_i`, i.e.
+    :math:`\mathbf{x}_i,i:v_i\in V`.
+
+    A GMRF is described by an undirected graph, where the vertexes stand for
+    random variables and the edges impose statistical constraints on these
+    random variables. Thus, the GMRF models the set of random variables with
+    a multivariate normal distribution
+
+    .. math::
+       p(X=\mathbf{x}|G)\sim\mathcal{N}(\boldsymbol{\mu},\boldsymbol{\Sigma})
+
+    We denote by :math:`\mathbf{Q}` the block-sparse precision matrix that is
+    the inverse of the covariance matrix :math:`\boldsymbol{\Sigma}`, i.e.
+    :math:`\mathbf{Q}=\boldsymbol{\Sigma}^{-1}`.  By applying the GMRF we make
+    the assumption that the random variables satisfy the three Markov
+    properties (pairwise, local and global) and that the blocks of the
+    precision matrix that correspond to disjoint vertexes are zero, i.e.
+
+    .. math::
+       \mathbf{Q}_{ij}=\mathbf{0}_{k\times k},\forall i,j:(v_i,v_j)\notin E
+
+    References
+    ----------
+    .. [1] H. Rue, and L. Held. "Gaussian Markov random fields: theory and
+       applications," CRC Press, 2005.
+    .. [2] E. Antonakos, J. Alabort-i-Medina, and S. Zafeiriou. "Active
+       Pictorial Structures", IEEE International Conference on Computer Vision
+       & Pattern Recognition (CVPR), Boston, MA, USA, pp. 5435-5444, June 2015.
+    """
+    def __init__(self, samples, graph, mode='concatenation', n_components=None,
+                 single_precision=False, sparse=True, n_samples=None, bias=0,
```

Comment:
The only thing we need to check is the bug you are seeing with building single_precision models!